src/rgw/rgw_notify.cc

```diff
@@ -616,22 +616,25 @@ int remove_persistent_topic(const std::string& topic_name, optional_yield y) {
   return s_manager->remove_persistent_topic(topic_name, y);
 }
 
-rgw::sal::Object* get_object_with_atttributes(const req_state* s, rgw::sal::Object* obj) {
+rgw::sal::Object* get_object_with_atttributes(
+  const reservation_t& res, rgw::sal::Object* obj) {
   // in case of copy obj, the tags and metadata are taken from source
-  const auto src_obj = s->src_object ? s->src_object.get() : obj;
+  const auto src_obj = res.src_object ? res.src_object : obj;
   if (src_obj->get_attrs().empty()) {
     if (!src_obj->get_bucket()) {
-      src_obj->set_bucket(s->bucket.get());
+      src_obj->set_bucket(res.bucket);
     }
-    if (src_obj->get_obj_attrs(s->obj_ctx, s->yield, s) < 0) {
+    if (src_obj->get_obj_attrs(res.obj_ctx, res.yield, res.dpp) < 0) {
       return nullptr;
     }
   }
   return src_obj;
 }
 
-void metadata_from_attributes(const req_state* s, rgw::sal::Object* obj, KeyValueMap& metadata) {
-  const auto src_obj = get_object_with_atttributes(s, obj);
+static inline void metadata_from_attributes(
+  const reservation_t& res, rgw::sal::Object* obj) {
+  auto metadata = res.x_meta_map;
```

Comment:
shouldn't this be a reference to the metadata in "res"?
otherwise, this is just updating the local variable.

Reply:
I'll check, was a merge thing

Reply:
ok, right;  I think I was going to remove the alias, but it's a bit nicer to read with it